src/shared/label-util.c

```diff
@@ -99,6 +99,24 @@ int mknod_label(const char *pathname, mode_t mode, dev_t dev) {
         return mac_smack_fix(pathname, 0);
 }
 
+int mknodat_label(int dirfd, const char *pathname, mode_t mode, dev_t dev) {
+        int r;
+
+        assert(pathname);
+
+        r = mac_selinux_create_file_prepare_at(dirfd, pathname, mode);
+        if (r < 0)
+                return r;
+
+        r = RET_NERRNO(mknodat(dirfd, pathname, mode, dev));
```

Comment:
I.e., please conditionalize the use of `mknodat` and `mknod` based on `dirfd >= 0`